File: src/main/java/com/negosud/api/service/InventaireService.java
package com.negosud.api.service;

import com.negosud.api.dto.request.InventaireRequestDto;
import com.negosud.api.dto.response.InventaireResponseDto;

import java.util.List;

public interface InventaireService {

    List<InventaireResponseDto> findAll();

    InventaireResponseDto findById(Long id);

    InventaireResponseDto create(InventaireRequestDto dto);

    /**
     * Applique les quantités constatées comme nouvelles quantités en stock.
     * Un inventaire ne peut être régularisé qu'une seule fois.
     */
    InventaireResponseDto regulariser(Long id);
}

File: front/src/services/inventaireService.js
import api from './api';

const BASE = '/inventaires';

export const inventaireService = {
  findAll: () => api.get(BASE),
  findById: (id) => api.get(`${BASE}/${id}`),
  create: (data) => api.post(BASE, data),
  regulariser: (id) => api.post(`${BASE}/${id}/regulariser`),
};
